### ISSUE TITLE
build of gnome2.gtkglextmm failed

### ISSUE THREAD

**Issue by Lassulus** [NixOS/nixpkgs#8327]
I tried building gtkglextmm, it failed with:
```
Package 'pangox', required by 'gdkglext-x11-1.0', not found
```
Then I imported `pangox_compat` into the default.nix + nativeBuildInputs / propagatedBuildInputs.
This time it failed with:
```
checking for GtkGLExt - version >= 1.0.0... no
*** Could not run GtkGLExt test program, checking why...
*** The test program failed to compile or link. See the file config.log for the
*** exact error that occured. This usually means GtkGLExt is incorrectly installed.
configure: error: GtkGLExt not found
```

I will add the config.log and my patched default.nix .

**command**
```
 nix-build -K -E 'with import <nixpkgs> { }; callPackage ./default.nix { inherit (gnome2) gtkglext;}'
```
in nixpkgs/pkgs/desktops/gnome-2/platform/gtkglextmm (original default.nix)

**output**
```
these derivations will be built:
  /nix/store/qxqjrg7d2bd53scjf9hq3y231fv8n15p-gtkglextmm-1.2.0.drv
building path(s) ‘/nix/store/9clzb6dkx259x68pmc0fa6micvckk3qv-gtkglextmm-1.2.0’
unpacking sources
unpacking source archive /nix/store/dmmxcddcybmmq53mzzpp6f68zf0izyfb-gtkglextmm-1.2.0.tar.bz2
source root is gtkglextmm-1.2.0
patching sources
applying patch /nix/store/iivg9zsm9r513yxns52i9v599vcvnq43-gdk.patch
patching file gtkglext/gtkmm/gl/widget.cc
patch unexpectedly ends in middle of line
Hunk #1 succeeded at 17 with fuzz 1.
configuring
fixing libtool script ./ltmain.sh
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/9clzb6dkx259x68pmc0fa6micvckk3qv-gtkglextmm-1.2.0  
checking for a BSD-compatible install... /nix/store/bnlcsbw7nhd21vw4xssnxwi9y7h2asjj-coreutils-8.23/bin/install -c
checking whether build environment is sane... yes
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking for gcc... gcc
checking for C compiler default output file name... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables... 
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ANSI C... none needed
checking for style of include used by make... GNU
checking dependency style of gcc... none
checking how to run the C preprocessor... gcc -E
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... none
checking how to run the C++ preprocessor... g++ -E
checking for strerror in -lcposix... no
checking for a sed that does not truncate output... /nix/store/50lmmxx9rynhbndaacwds8g99fgdjz7m-gnused-4.2.2/bin/sed
checking for egrep... grep -E
checking for ld used by gcc... /nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/ld
checking if the linker (/nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/ld) is GNU ld... yes
checking for /nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/ld option to reload object files... -r
checking for BSD-compatible nm... /nix/store/yxg5zsjxp1l16xk7r23rayyc18fkxynk-binutils-2.23.1/bin/nm -B
checking whether ln -s works... yes
checking how to recognise dependent libraries... pass_all
./configure: line 4767: /usr/bin/file: No such file or directory
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking how to run the C++ preprocessor... g++ -E
checking for g77... no
checking for f77... no
checking for xlf... no
checking for frt... no
checking for pgf77... no
checking for fort77... no
checking for fl32... no
checking for af77... no
checking for f90... no
checking for xlf90... no
checking for pgf90... no
checking for epcf90... no
checking for f95... no
checking for fort... no
checking for xlf95... no
checking for ifc... no
checking for efc... no
checking for pgf95... no
checking for lf95... no
checking for gfortran... no
checking whether we are using the GNU Fortran 77 compiler... no
checking whether  accepts -g... no
checking the maximum length of command line arguments... 32768
checking command to parse /nix/store/yxg5zsjxp1l16xk7r23rayyc18fkxynk-binutils-2.23.1/bin/nm -B output from gcc object... ok
checking for objdir... .libs
checking for ar... ar
checking for ranlib... ranlib
checking for strip... strip
checking for correct ltmain.sh version... yes
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC
checking if gcc PIC flag -fPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking whether the gcc linker (/nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/ld) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
configure: creating libtool
appending configuration tag "CXX" to libtool
checking for ld used by g++... /nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/ld
checking if the linker (/nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/ld) is GNU ld... yes
checking whether the g++ linker (/nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/ld) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC
checking if g++ PIC flag -fPIC works... yes
checking if g++ static flag -static works... yes
checking if g++ supports -c -o file.o... yes
checking whether the g++ linker (/nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
appending configuration tag "F77" to libtool
checking for some Win32 platform... no
checking for native Win32... no
checking whether build environment is sane... yes
checking for gcc option to accept ANSI C... none needed
checking for a BSD-compatible install... /nix/store/bnlcsbw7nhd21vw4xssnxwi9y7h2asjj-coreutils-8.23/bin/install -c
checking whether make sets $(MAKE)... (cached) yes
checking whether ln -s works... yes
checking for gawk... (cached) gawk
checking for gm4... no
checking for m4... no
checking whether m4 is GNU m4... no
checking whether make is GNU Make... yes
checking for compiler warning flags to use... -Wall
checking for pkg-config... /nix/store/qlmksvpzkj7xc6kwi25z35fqssff6nwr-pkg-config-0.28/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for BASE_DEPENDENCIES... configure: error: Package requirements (gtkglext-1.0 >= 1.0.0 gdkglext-1.0 >= 1.0.0 gtkmm-2.4 >= 2.4.0 gdkmm-2.4 >= 2.4.0 pangomm-1.4 >= 2.4.0 ) were not met:

Package 'pangox', required by 'gdkglext-x11-1.0', not found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables BASE_DEPENDENCIES_CFLAGS
and BASE_DEPENDENCIES_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.

note: keeping build directory ‘/tmp/nix-build-gtkglextmm-1.2.0.drv-12’
builder for ‘/nix/store/qxqjrg7d2bd53scjf9hq3y231fv8n15p-gtkglextmm-1.2.0.drv’ failed with exit code 1
```

======-----
**command**
```
 nix-build -K -E 'with import <nixpkgs> { }; callPackage ./default.nix { inherit (gnome2) gtkglext;}'
```
in nixpkgs/pkgs/desktops/gnome-2/platform/gtkglextmm (patched default.nix)
**default.nix**
```
{ stdenv, fetchurl, pkgconfig, gtkglext, gtkmm, gtk, mesa, gdk_pixbuf, pango, pangox_compat }:

stdenv.mkDerivation rec {
  name = "gtkglextmm-${minVer}.0";
  minVer = "1.2";

  src = fetchurl {
    url = "mirror://gnome/sources/gtkglextmm/${minVer}/${name}.tar.bz2";
    sha256 = "6cd4bd2a240e5eb1e3a24c5a3ebbf7ed905b522b888439778043fdeb58771fea";
  };

  #patches = [ ./gdk.patch ];

  nativeBuildInputs = [ pkgconfig ];

  propagatedBuildInputs = [ pango pangox_compat gtkglext gtkmm gtk mesa gdk_pixbuf ];

  meta = {
    description = "C++ wrappers for GtkGLExt";

    license = stdenv.lib.licenses.lgpl2Plus;

    platforms = stdenv.lib.platforms.linux;
  };
}
```

**output**
```
these derivations will be built:
  /nix/store/sh32kp9cdpb6bk2w65ja2d6vrcr3jgra-gtkglextmm-1.2.0.drv
building path(s) ‘/nix/store/fy0dlvk0pqlabh6dkb0f7vfc3l4p2b50-gtkglextmm-1.2.0’
unpacking sources
unpacking source archive /nix/store/dmmxcddcybmmq53mzzpp6f68zf0izyfb-gtkglextmm-1.2.0.tar.bz2
source root is gtkglextmm-1.2.0
patching sources
configuring
fixing libtool script ./ltmain.sh
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/fy0dlvk0pqlabh6dkb0f7vfc3l4p2b50-gtkglextmm-1.2.0  
checking for a BSD-compatible install... /nix/store/bnlcsbw7nhd21vw4xssnxwi9y7h2asjj-coreutils-8.23/bin/install -c
checking whether build environment is sane... yes
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking for gcc... gcc
checking for C compiler default output file name... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables... 
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ANSI C... none needed
checking for style of include used by make... GNU
checking dependency style of gcc... none
checking how to run the C preprocessor... gcc -E
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... none
checking how to run the C++ preprocessor... g++ -E
checking for strerror in -lcposix... no
checking for a sed that does not truncate output... /nix/store/50lmmxx9rynhbndaacwds8g99fgdjz7m-gnused-4.2.2/bin/sed
checking for egrep... grep -E
checking for ld used by gcc... /nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/ld
checking if the linker (/nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/ld) is GNU ld... yes
checking for /nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/ld option to reload object files... -r
checking for BSD-compatible nm... /nix/store/yxg5zsjxp1l16xk7r23rayyc18fkxynk-binutils-2.23.1/bin/nm -B
checking whether ln -s works... yes
checking how to recognise dependent libraries... pass_all
./configure: line 4767: /usr/bin/file: No such file or directory
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking how to run the C++ preprocessor... g++ -E
checking for g77... no
checking for f77... no
checking for xlf... no
checking for frt... no
checking for pgf77... no
checking for fort77... no
checking for fl32... no
checking for af77... no
checking for f90... no
checking for xlf90... no
checking for pgf90... no
checking for epcf90... no
checking for f95... no
checking for fort... no
checking for xlf95... no
checking for ifc... no
checking for efc... no
checking for pgf95... no
checking for lf95... no
checking for gfortran... no
checking whether we are using the GNU Fortran 77 compiler... no
checking whether  accepts -g... no
checking the maximum length of command line arguments... 32768
checking command to parse /nix/store/yxg5zsjxp1l16xk7r23rayyc18fkxynk-binutils-2.23.1/bin/nm -B output from gcc object... ok
checking for objdir... .libs
checking for ar... ar
checking for ranlib... ranlib
checking for strip... strip
checking for correct ltmain.sh version... yes
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC
checking if gcc PIC flag -fPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking whether the gcc linker (/nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/ld) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
configure: creating libtool
appending configuration tag "CXX" to libtool
checking for ld used by g++... /nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/ld
checking if the linker (/nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/ld) is GNU ld... yes
checking whether the g++ linker (/nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/ld) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC
checking if g++ PIC flag -fPIC works... yes
checking if g++ static flag -static works... yes
checking if g++ supports -c -o file.o... yes
checking whether the g++ linker (/nix/store/ylj1q31a7x3pvcraxk60dyv9y4wmwcrc-gcc-wrapper-4.8.4/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
appending configuration tag "F77" to libtool
checking for some Win32 platform... no
checking for native Win32... no
checking whether build environment is sane... yes
checking for gcc option to accept ANSI C... none needed
checking for a BSD-compatible install... /nix/store/bnlcsbw7nhd21vw4xssnxwi9y7h2asjj-coreutils-8.23/bin/install -c
checking whether make sets $(MAKE)... (cached) yes
checking whether ln -s works... yes
checking for gawk... (cached) gawk
checking for gm4... no
checking for m4... no
checking whether m4 is GNU m4... no
checking whether make is GNU Make... yes
checking for compiler warning flags to use... -Wall
checking for pkg-config... /nix/store/qlmksvpzkj7xc6kwi25z35fqssff6nwr-pkg-config-0.28/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for BASE_DEPENDENCIES... yes
checking for pkg-config... /nix/store/qlmksvpzkj7xc6kwi25z35fqssff6nwr-pkg-config-0.28/bin/pkg-config
checking for GtkGLExt - version >= 1.0.0... no
*** Could not run GtkGLExt test program, checking why...
*** The test program failed to compile or link. See the file config.log for the
*** exact error that occured. This usually means GtkGLExt is incorrectly installed.
configure: error: GtkGLExt not found
builder for ‘/nix/store/sh32kp9cdpb6bk2w65ja2d6vrcr3jgra-gtkglextmm-1.2.0.drv’ failed with exit code 1
error: build of ‘/nix/store/sh32kp9cdpb6bk2w65ja2d6vrcr3jgra-gtkglextmm-1.2.0.drv’ failed
```

**config.log**
https://gist.github.com/Lassulus/623432880469f59c0ae1#file-config-log

**Comment by bjornfor** [comment]
What do you need gtkglextmm for? I discovered the same error a while back when I tried to package vpython (which depends on gtkglextmm).

My WIP branch is here: https://github.com/bjornfor/nixpkgs/tree/vpython2

Hm, it seems gtkglext 1.2.0 from 2006 is the newest version. Dead project?

**Comment by bjornfor** [comment]
In my branch, configure succeeds but the build fails because it doesn't find glib (even though it's an input).

**Comment by bjornfor** [comment]
Looks like debian might have some fixes for gtkglextmm...

**Comment by bjornfor** [comment]
Yes they do. Will push fix soon.

**Comment by bjornfor** [comment]
gtkglextmm is fixed in master (903183d9e467c2bfe9c102f28328cbff2c95fe19).

**Comment by bjornfor** [comment]
And now backported to release-14.12.
